gsl_simulation/services/simulation_projet_service.py

```diff
@@ -1,33 +1,127 @@
 from decimal import Decimal
 
-from gsl_simulation.models import SimulationProjet
+from gsl_core.models import Perimetre
+from gsl_projet.models import Projet
+from gsl_projet.services import ProjetService
+from gsl_simulation.models import Simulation, SimulationProjet
 
 
 class SimulationProjetService:
+    @classmethod
+    def create_or_update_simulation_projets_from_projet(cls, projet: Projet):
+        simulation_projets = SimulationProjet.objects.filter(projet=projet)
+        for simulation_projet in simulation_projets:
+            cls.create_or_update_simulation_projet_from_projet(
+                projet, simulation_projet.simulation_id
+            )
+
+    @classmethod
+    def create_or_update_simulation_projet_from_projet(
+        cls, projet: Projet, simulation_id: int
+    ):
+        try:
+            simulation_projet = SimulationProjet.objects.get(
+                projet=projet, simulation_id=simulation_id
+            )
+        except SimulationProjet.DoesNotExist:
+            simulation = Simulation.objects.get(id=simulation_id)
+            simulation_projet = SimulationProjet(
+                projet=projet,
+                enveloppe_id=simulation.enveloppe_id,
+                simulation_id=simulation_id,
+            )
+        montant = cls.get_initial_montant_from_projet(projet)
+        simulation_projet.taux = (
+            projet.dossier_ds.annotations_taux
+            or ProjetService.compute_taux_from_montant(projet, montant)
+        )
+        simulation_projet.montant = montant
+        simulation_projet.status = cls.get_simulation_projet_status(projet)
+        simulation_projet.save()
+
```

Comment:
Ici aussi, est-ce qu'un `update_or_create` nous faciliterait la vie ?

Reply:
Un truc du genre : (pas du tout testé, tapé dans l'interface de Github)
```suggestion
        simulation_projet, _ = SimulationProjet.objects.update_or_create(
            projet=projet, simulation_id=simulation_id,
            defaults={
                "enveloppe_id":simulation.enveloppe_id, # peut-être pas besoin ?
                "montant":cls.get_initial_montant_from_projet(projet),
                "taux": (
                    projet.dossier_ds.annotations_taux
                    or ProjetService.compute_taux_from_montant(projet, montant)
                ),
                "status": cls.get_simulation_projet_status(projet)
            }
        )
```

Reply:
Ah oui pas mal !
Je teste ça de suite !